mysql-test/suite/period/r/delete.result

```diff
@@ -0,0 +1,171 @@
+create trigger tr1ins before insert on t for each row call log('>INS');
+select * from t;
+id	s	e
+2	1999-01-01	2000-01-01
+3	2018-01-01	2018-12-12
+# generated columns are updated
+create or replace table t (s date, e date,
+xs date as (s) stored, xe date as (e) stored,
+period for apptime(s, e));
+insert into t values('1999-01-01', '2018-12-12', default, default);
+select * from t;
+s	e	xs	xe
+1999-01-01	2018-12-12	1999-01-01	2018-12-12
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select * from t;
+s	e	xs	xe
+1999-01-01	2000-01-01	1999-01-01	2000-01-01
+2018-01-01	2018-12-12	2018-01-01	2018-12-12
+truncate t;
+# same for trigger case
+insert into t values('1999-01-01', '2018-12-12', default, default);
+create trigger tr1ins before insert on t for each row call log('>INS');
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select * from t;
+s	e	xs	xe
+1999-01-01	2000-01-01	1999-01-01	2000-01-01
+2018-01-01	2018-12-12	2018-01-01	2018-12-12
+# system-time columns are updated
+create or replace table t (s date, e date,
+period for apptime(s, e)) with system versioning;
+set @@timestamp=3141592;
+insert into t values('1999-01-01', '2018-12-12');
+insert into t values('1999-01-01', '1999-12-12');
+select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	2018-12-12	3141592.000000	2147483647.999999
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+set @@timestamp=314159265.358979;
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+# same for trigger case
+delete from t;
+delete history from t;
+set @@timestamp=3141592;
+insert into t values('1999-01-01', '2018-12-12');
+insert into t values('1999-01-01', '1999-12-12');
+select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+1999-01-01	2018-12-12	3141592.000000	2147483647.999999
+set @@timestamp=314159265.358979;
+delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
+select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+select *, unix_timestamp(row_start),
+unix_timestamp(row_end) from t for system_time all;
+s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
+1999-01-01	2018-12-12	3141592.000000	314159265.358979
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+create or replace database test;
```

Comment:
## Bug 10: `period.delete` fails

After this modification `period.delete` fails:
```diff
diff --git a/mysql-test/suite/period/t/delete.test b/mysql-test/suite/period/t/delete.test
index 9fca55a0d0d..12a4fced673 100644
--- a/mysql-test/suite/period/t/delete.test
+++ b/mysql-test/suite/period/t/delete.test
@@ -1,4 +1,5 @@
-source suite/period/engines.inc;
+source suite/versioning/engines.inc;
+source suite/versioning/common.inc;
 source suite/period/log.inc;
 
 create or replace table t (id int, s date, e date, period for apptime(s,e));
```
Failure diff:
```diff
--- ./mysql-test/suite/period/r/delete.result	2018-09-04 12:41:42.528374817 +0300
+++ ./mysql-test/suite/period/r/delete.reject	2018-09-06 15:44:25.596917541 +0300
@@ -144,8 +144,8 @@
 delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
 select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
 s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
-1999-01-01	2000-01-01	314159265.358979	2147483647.999999
 1999-01-01	1999-12-12	3141592.000000	2147483647.999999
+2018-01-01	2018-12-12	314159265.358979	2147483647.999999
 # same for trigger case
 delete from t;
 delete history from t;
@@ -154,18 +154,18 @@
 insert into t values('1999-01-01', '1999-12-12');
 select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
 s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
-1999-01-01	1999-12-12	3141592.000000	2147483647.999999
 1999-01-01	2018-12-12	3141592.000000	2147483647.999999
+1999-01-01	1999-12-12	3141592.000000	2147483647.999999
 set @@timestamp=314159265.358979;
 delete from t for portion of apptime from '2000-01-01' to '2018-01-01';
 select *, unix_timestamp(row_start), unix_timestamp(row_end) from t;
 s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
 1999-01-01	1999-12-12	3141592.000000	2147483647.999999
-1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+2018-01-01	2018-12-12	314159265.358979	2147483647.999999
 select *, unix_timestamp(row_start),
 unix_timestamp(row_end) from t for system_time all;
 s	e	unix_timestamp(row_start)	unix_timestamp(row_end)
-1999-01-01	2018-12-12	3141592.000000	314159265.358979
+1999-01-01	2000-01-01	314159265.358979	314159265.358979
 1999-01-01	1999-12-12	3141592.000000	2147483647.999999
-1999-01-01	2000-01-01	314159265.358979	2147483647.999999
+2018-01-01	2018-12-12	314159265.358979	2147483647.999999
 create or replace database test;
```